cocos/base/StringPool.h

```diff
@@ -0,0 +1,131 @@
+/****************************************************************************
+ Copyright (c) 2020-2021 Xiamen Yaji Software Co., Ltd.
+
+ http://www.cocos.com
+
+ Permission is hereby granted, free of charge, to any person obtaining a copy
+ of this software and associated engine source code (the "Software"), a limited,
+ worldwide, royalty-free, non-assignable, revocable and non-exclusive license
+ to use Cocos Creator solely to develop games on your target platforms. You shall
+ not use Cocos Creator software for developing other software or tools that's
+ used for developing games. You are not granted to publish, distribute,
+ sublicense, and/or sell copies of Cocos Creator.
+
+ The software or tools in this License Agreement are licensed, not sold.
+ Xiamen Yaji Software Co., Ltd. reserves all rights not expressly granted to you.
+
+ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ THE SOFTWARE.
+****************************************************************************/
+
+#pragma once
+
+#include "StringHandle.h"
+#include "threading/ReadWriteLock.h"
+#include <assert.h>
+#include <cstring>
+#include <map>
+#include <vector>
+
+namespace cc {
+
+struct StringCompare final {
+    inline bool operator()(char const *lhs, char const *rhs) const noexcept {
+        return strcmp(lhs, rhs) < 0;
+    }
+};
+
+template <bool ThreadSafe>
+class StringPool final {
+public:
+    StringPool() noexcept          = default;
+    ~StringPool()                  = default;
+    StringPool(StringPool const &) = delete;
+    StringPool(StringPool &&)      = delete;
+    StringPool &operator=(StringPool const &) = delete;
+    StringPool &operator=(StringPool &&) = delete;
+
+    StringHandle StringToHandle(char const *const str) noexcept;
```

Comment:
`StringToHandle ` -> `stringToHandle `. Other functions of this class have the same issue.